src/main/java/emu/grasscutter/utils/StartupArguments.java

```diff
@@ -0,0 +1,128 @@
+package emu.grasscutter.utils;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import emu.grasscutter.BuildConfig;
+import emu.grasscutter.Grasscutter;
+import emu.grasscutter.net.packet.PacketOpcodesUtils;
+import emu.grasscutter.tools.Tools;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.function.Function;
+
+import static emu.grasscutter.config.Configuration.DATA;
+
+/**
+ * A parser for start-up arguments.
+ */
+public final class StartupArguments {
+    private StartupArguments() {
+        // This class is not meant to be instantiated.
+    }
+
+    /* A map of parameter -> argument handler. */
+    private static final Map<String, Function<String, Boolean>> argumentHandlers = Map.of(
+        "-handbook", StartupArguments::generateHandbook,
+        "-dumppacketids", parameter -> {
+            PacketOpcodesUtils.dumpPacketIds(); return true;
+        },
+        "-gachamap", StartupArguments::generateGachaMap,
+        "-version", StartupArguments::printVersion,
+        "-debug", StartupArguments::enableDebug,
+        "-lang", parameter -> {
+            Grasscutter.setPreferredLanguage(parameter); return false;
+        },
+
+        // Aliases.
+        "-v", StartupArguments::printVersion,
+        "-debugall", StartupArguments::enableDebug
```

Comment:
`-debugall` is not an alias for `-debug`, `-debug` currently sets the level to `Level.INFO` and `-debugall` to `Level.DEBUG`

Reply:
the first *five* times I look at it, I swore it was the same